sdk/containerregistry/azure-containerregistry/azure/containerregistry/_authentication_policy.py

```diff
@@ -19,13 +19,13 @@
 class ContainerRegistryChallengePolicy(HTTPPolicy):
     """Authentication policy for ACR which accepts a challenge"""
 
```

Comment:
You can put it here:

```py
_exchange_client: Union[AnonymousACRExchangeClient, ACRExchangeClient]
```

Reply:
If you intend to type self._exchange_client as one of `AnonymousACRExchangeClient` or `ACRExchangeClient` then I think you'll need to cast in the init. A type checker won't be able to resolve which client the attribute gets set to based on the runtime check for if self._credential is None. Funnily enough, it seems the Union type works out because both exchange clients implement the same interface